src/watchmaker/workers/salt.py

```diff
@@ -262,10 +273,47 @@ def _build_salt_formula(self, extract_dir):
                 salt_content_filename
             ))
             self.retrieve_file(self.salt_content, salt_content_file)
+            temp_extract_dir = extract_dir
+            if self.salt_content_path:
+                temp_extract_dir = os.sep.join((
+                    self.working_dir,
+                    'salt-archive'
+                ))
             self.extract_contents(
                 filepath=salt_content_file,
-                to_directory=extract_dir
+                to_directory=temp_extract_dir
```

Comment:
i see we're now extracting to `temp_extract_dir` in all cases but only *IF* `salt_content_path` is set do we move content back to `extract_dir`. when `salt_content_path` is not set, the content is not moved back. i think case needs to be handled.

also, it seems with these changes the function definition is becoming a leaky abstraction. it seems what we care about is not necessarily that we are extracting into `extract_dir`. or is it? is another function moving the extracted files _again_? might need to take a look at how this function is called, what sets `extract_dir` and maybe rename the argument to account for the actual usage?

Reply:
So, we start out with setting `temp_extract_dir = extract_dir` on line 276. That should handle the case you mention. 
`extract_dir` is essentially the `salt_srv` dir. In Linux, for e.g. this is '/srv/watchmaker/salt' where the salt files are extracted to. 
https://github.com/plus3it/watchmaker/blob/62400d9b27f6beac5a4af5d89daf7c072429ce87/src/watchmaker/workers/salt.py#L683
The archive is currently extracted directly into that dir. What this abstraction does, is in case where `salt_content_path` is set, the archive is extracted to the working directory, then from the glob are copied into the `salt_srv` a.k.a the `extract_dir` dir.

Reply:
> So, we start out with setting `temp_extract_dir = extract_dir` on line 276. That should handle the case you mention.

oops i missed that assignment. you're right! that pretty much addresses the entire prior comment. 😂  ok, so the only thing i don't like about this approach is that we test `if self.salt_content_path` twice. what if we reverse the logic here and then return immediately?

```suggestion
            if not self.salt_content_path:
                self.extract_contents(
                    filepath=salt_content_file,
                    to_directory=extract_dir,
                )
                return
            
            # now we can proceed with `salt_content_path` logic without any condition
```